thrust/thrust/detail/function.h

```diff
@@ -39,7 +39,7 @@ struct wrapped_function
 
   _CCCL_EXEC_CHECK_DISABLE
   template <typename... Ts>
-  _CCCL_FORCEINLINE _CCCL_HOST_DEVICE Result operator()(Ts&&... args) const
+  inline _CCCL_HOST_DEVICE Result operator()(Ts&&... args) const
```

Comment:
@miscco at least locally, this change avoids the gcc optimizer issue.

Reply:
That is awesome. I will test whether we can avoid that hack in the other tests too. Will file a separate PR though

Reply:
looks like this is still failing with gcc 12 😿 
https://github.com/NVIDIA/cccl/actions/runs/10967082905/job/30507674109?pr=2439

Reply:
it fixed almost all of the failures tho. just two remain. i can track those down when i get back from vacation.